Description: Use Debian gmp and cdd library packages instead of building them.
Author: Doug Torrance <dtorrance@debian.org>
Forwarded: not-needed
Last-Update: 2022-06-23

--- a/configure.ac
+++ b/configure.ac
@@ -18,10 +18,6 @@
 dnl AC_CHECK_LIB(stdc, main)
 dnl Replace `main' with a function in -lstdc++:
 dnl AC_CHECK_LIB(stdc++, main)
-dnl Make libgmp:
-make -C external gmp
-dnl Make libcddgmp:
-make -C external cdd
 dnl Check for soplex:
 AC_CHECK_LIB(soplex, main, , csh -c 'echo soplex not found: --soplex will not work.')
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -85,14 +85,11 @@
 
 LDADD           = ../lib-src/libTOPCOM.a \
                   ../lib-src-reg/libCHECKREG.a \
-                  ../external/lib/libcddgmp.a \
-                  ../external/lib/libgmpxx.a \
-                  ../external/lib/libgmp.a
+                  -lcddgmp -lgmpxx -lgmp
 
 AM_CPPFLAGS     = -I../lib-src
 AM_CPPFLAGS    += -I../lib-src-reg
 AM_CPPFLAGS    += -I../wrap-gmp-gmpxx
-AM_CPPFLAGS    += -I../external/include
 AM_CPPFLAGS    += -I$(includedir)
 
 AM_CPPFLAGS    += -DVERBOSE -DGMPRATIONAL -ftemplate-depth-30
--- a/lib-src-reg/Makefile.am
+++ b/lib-src-reg/Makefile.am
@@ -12,7 +12,7 @@
 
 AM_CPPFLAGS     = -I../lib-src 
 AM_CPPFLAGS    += -I../wrap-gmp-gmpxx
-AM_CPPFLAGS    += -I../external/include
+AM_CPPFLAGS    += -I/usr/include/cdd
 AM_CPPFLAGS    += -I$(includedir)
 
 AM_CPPFLAGS    += -DVERBOSE -DGMPRATIONAL -ftemplate-depth-30
--- a/src-reg/Makefile.am
+++ b/src-reg/Makefile.am
@@ -4,14 +4,12 @@
 
 LDADD           = ../lib-src-reg/libCHECKREG.a \
                   ../lib-src/libTOPCOM.a \
-                  ../external/lib/libcddgmp.a \
-                  ../external/lib/libgmpxx.a \
-                  ../external/lib/libgmp.a
+                  -lcddgmp -lgmpxx -lgmp
 
 AM_CPPFLAGS     = -I../lib-src
 AM_CPPFLAGS    += -I../lib-src-reg
 AM_CPPFLAGS    += -I../wrap-gmp-gmpxx
-AM_CPPFLAGS    += -I../external/include
+AM_CPPFLAGS    += -I/usr/include/cdd
 AM_CPPFLAGS    += -I$(includedir)
 
 
--- a/wrap-gmp-gmpxx/Makefile.am
+++ b/wrap-gmp-gmpxx/Makefile.am
@@ -2,7 +2,6 @@
 Integer.h \
 Rational.h
 
-AM_CPPFLAGS        = -I../external/include
-AM_CPPFLAGS       += -I$(includedir)
+AM_CPPFLAGS       = -I$(includedir)
 
 AM_CXXFLAGS     = -O2
--- a/lib-src/Makefile.am
+++ b/lib-src/Makefile.am
@@ -90,7 +90,7 @@
 
 AM_CPPFLAGS     = -I../lib-src-reg
 AM_CPPFLAGS    += -I../wrap-gmp-gmpxx
-AM_CPPFLAGS    += -I../external/include
+AM_CPPFLAGS    += -I/usr/include/cdd
 AM_CPPFLAGS    += -I$(includedir)
 
 AM_CPPFLAGS    += -DVERBOSE -DGMPRATIONAL -ftemplate-depth-35
